content/docs/command-reference/import.md

```diff
@@ -104,9 +105,10 @@ repo at `url`) are not supported.
   > example below).
 
 - `--no-exec` - create `.dvc` file without actually downloading `url`. E.g. if
-  the file or directory already exist it can be used to skip download.
-  `dvc commit <out>.dvc` should be used to calculate the URL and data hash,
-  update the `.dvc` files, and save existing data to the cache.
+  the file or directory already exists, this can be used to skip the download.
+  The data hash is not calculated by this since only the metadata is saved into
```

Comment:
@efiop when we import objects do we trust the external source for the hash? if not, does it makes sense to do this?  It seems to me that `--no-exec` at least in the case of `import` can write hash as well if the goal is to avoid data download. WDYT?

Reply:
It is more complicated than that, but yeah, we trust it indirectly by trusting the cache. This is because we don't really `download`, but rather `pull + checkout`. We don't fill it on `--no-exec` for the `run` symmetry, but yeah, could consider it, though that has itsown implications. Let's leave it as is for now, since it is not a high priority.

Reply:
>  but rather pull + checkout

hmm, how do we do `pull` if hash is not set? I thought that we don't do anything with data at all - kinda create .dvc with some meta in it and that it.

if we do `pull` + `checkout` what happens if data already exists in workspace and is in cache, or exists and not in cache?

> Let's leave it as is for now, since it is not a high priority. 

PR is merged, yep.

Reply:
Pretty late to this conversation but just wanted to double check that with --no-exec there's no connection to the data source, right? Because I'm reviewing this text for an upcoming docs update. Thanks